src/test/java/teammates/test/cases/search/InstructorSearchTest.java

```diff
@@ -0,0 +1,133 @@
+
+        results = instructorsDb.searchInstructorsInWholeSystem("InStRuCtOr2");
+        verifySearchResults(results, ins2InCourse1, ins2InCourse2);
+
+        ______TS("success: search for instructors in whole system; instructors in archived courses should be included");
+
+        results = instructorsDb.searchInstructorsInWholeSystem("archived");
+        verifySearchResults(results, insInArchivedCourse);
+
+        ______TS("success: search for instructors in whole system; instructors in unregistered course should be included");
+
+        results = instructorsDb.searchInstructorsInWholeSystem("instructor5");
+        verifySearchResults(results, insInUnregCourse);
+
+        ______TS("success: search for instructors in whole system; instructors should be searchable by course id");
+        
+        results = instructorsDb.searchInstructorsInWholeSystem("idOfUnregisteredCourse");
+        verifySearchResults(results, insInUnregCourse);
+        
+        ______TS("success: search for instructors in whole system; instructors should be searchable by course name");
+        
+        results = instructorsDb.searchInstructorsInWholeSystem("idOfTypicalCourse2");
+        verifySearchResults(results, ins1InCourse2, ins2InCourse2, ins3InCourse2);
+        
+        ______TS("success: search for instructors in whole system; instructors should be searchable by their name");
+        
+        results = instructorsDb.searchInstructorsInWholeSystem("\"Instructor 5 of CourseNoRegister\"");
+        verifySearchResults(results, insInUnregCourse);
+        
+        ______TS("success: search for instructors in whole system; instructors should be searchable by their email");
+        
+        results = instructorsDb.searchInstructorsInWholeSystem("instructor2@course2.tmt");
+        verifySearchResults(results, ins2InCourse2);
+        
+        ______TS("success: search for instructors in whole system; instructors should be searchable by their google id");
+        
+        results = instructorsDb.searchInstructorsInWholeSystem("idOfInstructor5");
+        verifySearchResults(results, insInUnregCourse);
+        
+        ______TS("success: search for instructors in whole system; instructors should be searchable by their role");
+        
+        results = instructorsDb.searchInstructorsInWholeSystem("Manager");
+        verifySearchResults(results, ins2InCourse1);
+        
+        ______TS("success: search for instructors in whole system; deleted instructors no longer searchable");
+        
+        instructorsDb.deleteInstructor(ins1InCourse1.courseId, ins1InCourse1.email);
+        results = instructorsDb.searchInstructorsInWholeSystem("instructor1");
+        verifySearchResults(results, ins1InCourse2);
+        
+        ______TS("success: search for instructors in whole system; instructors created without searchability unsearchable");
```

Comment:
This test case is valid.
But there is a missing test case: Delete one instructor without delete the document, instructor won't be searchable and the document will be deleted in searching. 
You can achieve this by using `deleteEntity` in `InstructorsDb`.